### PR TITLE
Clean up old DocumentationNode/BasePageRenderer infrastructure

### PR DESCRIPTION
**Pull request by pgonzal**
This PR cleans up old constructs from the earlier **api-documenter** prototype, so that the Markdown and YAML documentation targets are both implemented using the new `DocItemSet` system.

### CODE REVIEW

File: libraries/api-documenter/src/markdown/MarkdownGenerator.ts [hunk 1]
```diff
@@ -21,48 +22,49 @@ import {
...
 
-  public writeDocs(renderer: BasePageRenderer): void {
-    console.log(os.EOL + `Deleting old *${renderer.outputFileExtension} files...` + os.EOL);
-    renderer.deleteOutputFiles();
+  public generateFiles(outputFolder: string): void {
+    this._outputFolder = outputFolder;
 
-    for (const apiJsonFile of this._apiJsonFiles) {
-      this._writePackagePage(apiJsonFile, renderer);
+    console.log();
```
Comment (nickpape-msft) [inline libraries/api-documenter/src/markdown/MarkdownGenerator.ts]: >console.log(); [](start = 4, length = 14)

keep this?

File: libraries/api-documenter/src/markdown/MarkdownGenerator.ts [hunk 2]
```diff
@@ -210,45 +203,63 @@ export class Documenter {
...
+          const constructorTitle: MarkupBasicElement[] = [
+            MarkupBuilder.createApiLink(
+              [MarkupBuilder.createCode(RenderingHelpers.getConciseSignature(docMember.name, apiMember), 'javascript')],
+              docMember.getApiReference())
+          ];
+
+          methodsTable.rows.push(
+            MarkupBuilder.createTableRow([
+              constructorTitle,
+              [],
```
Comment (nickpape-msft) [inline libraries/api-documenter/src/markdown/MarkdownGenerator.ts]: >[], [](start = 14, length = 3)

do you get the accessmodifier?

File: libraries/api-documenter/src/markdown/MarkdownGenerator.ts (hunk 1 above)
Comment (pgonzal) [reply libraries/api-documenter/src/markdown/MarkdownGenerator.ts]: This skips a line in the console output

---
In reply to: [141506540](https://github.com/Microsoft/web-build-tools/pull/381#discussion_r141506540) [](ancestors = 141506540)

File: libraries/api-documenter/src/markdown/MarkdownGenerator.ts (hunk 2 above)
Comment (pgonzal) [reply libraries/api-documenter/src/markdown/MarkdownGenerator.ts]: IApiConstructor currently doesn't support the access modifier.  It's one of the *.api.json improvements that will get fixed in an upcoming PR.

---
In reply to: [141506814](https://github.com/Microsoft/web-build-tools/pull/381#discussion_r141506814) [](ancestors = 141506814)

**Pull request merged**